File: pergola/kiwifruit_locations.rb
#!/usr/bin/env ruby

# Module for creating locations of kiwifruit (or flowers)

module Kiwifruit_locations
z_diff = 0.2
def getPose(number, x_max, y_max, z_mean)
	if number < 2
		anchors = 1
	else
		anchors = rand(1..((number/2).round))
	end
	fruit_locations_ = Array.new(number) { Array.new(3) }
	for i in (0, number*3).step(3)
		fruit_locations_[i][0] = rand(x_max)
		fruit_locations_[i][1] = rand(y_max)
		fruit_locations_[i][2] = z_mean+rand(-0.2,0.2)
	end
	return fruit_locations_

end

File: mybot_gazebo/worlds/empy/kiwifruit_locations.py
#!/usr/bin/env python

# This file contains the function used to get the randomised locations of fruit in the orchard. This is used by the em.py template.
import numpy as np
from random import randrange, uniform

def getFruitLocations(number, x_max, y_max, z_mean):
	z_diff = 0.2
	if number < 2:
		anchors = 1
	else:
		anchors = randrange(0,number/2)
	frand = uniform(0,10)
	fruit_locations_ = np.zeros((number, 3))
	for i in np.arange(0, number*3, 3):
		np.put(fruit_locations_, [i, i+1, i+2], [uniform(0,x_max), uniform(0,y_max), z_mean+uniform(-0.2,0.2)])
	return fruit_locations_
